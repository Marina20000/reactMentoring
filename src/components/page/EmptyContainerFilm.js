import React, {PureComponent} from 'react';
import style from './style.css';

export default class EmptyContainerFilm extends PureComponent{

    render() {
        return <div className={style.emptyResult}>Find your movie</div>;
    }
}