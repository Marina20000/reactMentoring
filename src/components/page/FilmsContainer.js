import React from 'react';
import Flexbox from 'flexbox-react';
import MovieItem from './../movieItems/MovieItem';

export default class FilmsContainer extends React.Component {
  constructor(props) {
    super(props);
  }


  render() {
    let images = [];
    let movies = !!this.props.movies?this.props.movies:[];
    movies.forEach(index => {
      images.push(<MovieItem params={index} key={index.id} />)
    })
    return (<Flexbox flexDirection="column" minHeight="100vh" width = "100%">
      <Flexbox element="header" height="60px">
      </Flexbox>
      <Flexbox flexGrow={1} flexWrap={'wrap'} justifyContent={'center'}>
        {images}
      </Flexbox>
      <Flexbox element="footer" height="60px">
      </Flexbox>
    </Flexbox>);
  };
}
