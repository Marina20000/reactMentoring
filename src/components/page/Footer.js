import React from 'react';
import Inscription from './../captions/Inscription';
import style from './style.css';

export default class Footer extends React.Component {

    render() {
        return <div style={{ height: 100, backgroundColor: 'black', bottom: 0 }}>
            <Inscription className={ style.netflix } inscription='netflixroulette'/>
       </div>;
    }

}