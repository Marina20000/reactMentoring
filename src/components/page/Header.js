import React from 'react';
import SearchPanel from './../textboxes/SearchPanel';
import Inscription from './../captions/Inscription';
import CreateButton from './../buttons/CreateButton';
import style from './style.css';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './../../actions/action';
import SearchResultPanel from './SearchResultPanel';


export class Header extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            title: true,
            genre: false,
            value: '',
            releaseDate: true,
            raiting: false
        };
    }

    toggleButton = (filter) => {
        this.setState({ title: !this.state.title, genre: !this.state.genre });
    };

    searchFilms = () => {
        let filter = this.state.title ? 'title':'genre';
        let sorting =this.state.releaseDate ? 'release_date':'raiting';
        this.props.actions.getData(this.request(filter,sorting));
    }

    request = (filter,sorting) => {
        return 'limit=20&search='+this.state.value+'&searchBy='+filter+'&sortBy='+sorting+'&sortOrder=desc';
    }

    handleChange = e => {
        this.setState({ value: e.target.value });
    }

    toggleSortings = () => {
        this.setState({ releaseDate: !this.state.releaseDate, raiting: !this.state.raiting });
    };

    render() {
        return (
            <div style={{ height: 500, backgroundImage: `url('\./image.jpg')`, backgroundSize: '100%' }}>
                <div className={style.container}>
                    <Inscription className={style.netflix} inscription='netflixroulette' />
                    <Inscription className={style.findYourMovie} inscription='FIND YOUR MOVIE' />
                    <SearchPanel onChange={this.handleChange} value={this.state.value}/>
                    <div style={{ width: '100%' }}>
                        <Inscription className={style.searchBy} inscription='SEARCH BY' />
                        <CreateButton onClick={() => { this.toggleButton('title') }} className={this.state.title ? style.selected : style.noneSelected} inscription='TITLE' />
                        <CreateButton onClick={() => { this.toggleButton('genre') }} className={this.state.genre ? style.selected : style.noneSelected} inscription='GENRE' />
                        <CreateButton onClick={() => { this.searchFilms() }} className={style.searchButton} inscription='SEARCH' />
                    </div>
                </div>
                <SearchResultPanel quantity={ !!this.props.movies ? this.props.movies.length : 0 } showPage2 = { this.props.showPage2 } toggleSortings={this.toggleSortings}
                raiting = { this.state.raiting } releaseDate = { this.state.releaseDate }/>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        movies: state.app.getFilms.data,
    }
  }
  
  function mapDispatchToProps(dispatch) {
    return {
      actions: bindActionCreators(actions, dispatch)
    }
  }
  
  export default connect(mapStateToProps, mapDispatchToProps)(Header);