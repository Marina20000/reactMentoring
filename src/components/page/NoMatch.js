import React from 'react';
import style from './style.css';

export default class NoMatch extends React.Component{
render() {
     return <div className={style.emptyResult}>
        Not found
    </div>;
}
}