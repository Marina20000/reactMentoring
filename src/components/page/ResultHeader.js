import React from 'react';
import MovieItem from './../movieItems/MovieImage';
import ContainerDescription from './../movieItems/ContainerDescription';
import Inscription from './../captions/Inscription';
import CreateButton from './../buttons/CreateButton';
import style from './style.css';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './../../actions/action';


export class ResultHeader extends React.Component {
    constructor(props) {
        super(props);
    }

    returnButtonClick = () => {
        this.props.actions.changePage();
    }

    render() {
        return (
            <div style={{ backgroundImage: `url('\./image.jpg')`, backgroundSize: '100%' }}>
                <div className={style.container}>
                    <Inscription className={style.netflix} inscription='netflixroulette' />
                    <CreateButton onClick={() => { this.returnButtonClick() }} className={style.returnButton} inscription='SEARCH' />
                    <img src={!!this.props.film ? this.props.film.poster_path : ''} className={style.movieImage} />
                    <ContainerDescription film={this.props.film} />

                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    console.log('state', state)
    return { film: state.clickFilm.film }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ResultHeader);