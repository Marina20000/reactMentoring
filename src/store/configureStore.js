import { createStore, applyMiddleware } from 'redux';
//const {} createStore } = Redux;
//var createStore = Redux.createStore;
import rootReducer from './../rootReducer';
import thunk from 'redux-thunk';


/**
 * Logs all actions and states after they are dispatched.
 */
const logger = store => next => action => {
    console.group(action.type)
    console.info('dispatching', action)
    let result = next(action)
    console.log('next state', store.getState())
    console.groupEnd()
    return result
  }
  
   // Sends crash reports as state is updated and listeners are notified.

  // const promiseExtractMiddleware = store => next => action => {
  //   if (action.type !== "PROMISE_EXTRACT") {
  //     return next(action)
  //   }

  //   const {
  //     data,
  //     types: [eventStart, eventSuccess, eventFallback],
  //   } = action.payload;

  //   store.dispatch({
  //     type: eventStart
  //   })

  //   data.then((response) => {
  //     store.dispatch({
  //       type: eventSuccess,
  //       payload: response
  //     })
  //   }, (error) => {
  //     store.dispatch({
  //       type: eventFallback,
  //       payload: error
  //     })
  //   })
  // }

export default function configureStore(initialState) {
    const store = createStore(rootReducer, initialState, applyMiddleware(thunk,logger
     // , promiseExtractMiddleware
    ));
    return store;
}



//const counter = (state = 0, action) => {
//switch (action.type) {
//case 'INCREMENT':
// return state+1;
//case 'DECREMENT':
//return state-1;
//default: return state;
//}
//}
//
//const { createStore } = Redux;
//const store = createStore (counter);

//const render = () => {
//document.body.innerText = store.getState();
//}

//store.subscribe(render);
//render();
//document.addEventListener('click', () => {
//store.dispatch ({type:'INCREMENT'});
//})
