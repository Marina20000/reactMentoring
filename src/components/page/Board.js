import React from 'react';
import FilmsContainer from './FilmsContainer';
import SearchResultPanel from './SearchResultPanel';
import Inscription from './../captions/Inscription';
import style from './style.css';

export default class Board extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        let searchResult = this.props.searchResult;
         return <div >
            { searchResult && <Inscription className={ style.noFilmsFound } inscription = 'No films found'/> }
            <FilmsContainer  movies = { this.props.movies }/>
        </div>;
    }
}