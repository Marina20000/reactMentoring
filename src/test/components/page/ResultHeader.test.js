import React from "react";
import renderer from 'react-test-renderer';
import { ResultHeader } from './../../../components/page/ResultHeader';
import reducer from './../../../reducers/reducers';
import configureStore from 'redux-mock-store';

describe('ResultHeader', () => {
    const mockStore = configureStore();
    let initialState  = {};
    let _props, store;
    it("testing with snapshot", () => {
        store = mockStore(initialState);
        _props={ store: store, data:[], film: {} }
        const tree = renderer.create(<ResultHeader {..._props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})