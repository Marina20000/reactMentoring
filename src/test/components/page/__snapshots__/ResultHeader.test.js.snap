// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultHeader testing with snapshot 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url('./image.jpg')",
      "backgroundSize": "100%",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="netflix"
    >
      netflixroulette
    </div>
    <button
      className="returnButton"
      onClick={[Function]}
    >
      SEARCH
    </button>
    <img
      className="movieImage"
      src={undefined}
    />
    <div
      className="containerDescription"
    >
      <div
        className="movieTitle"
      />
      <div
        className="movieYear"
      />
      <div
        className="movieYear"
      >
        undefined  min
      </div>
      <div
        className="description"
      />
    </div>
  </div>
</div>
`;
